src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py

```diff
@@ -42,21 +46,43 @@ def __init__(self, span: SpanEntity, collection_id: str, created_by: dict) -> No
         self.collection_id = collection_id
         self.id = span.span_id
         self.created_by = created_by
+        self.external_event_data_uris = []
 
-    def persist(self, client):
+    def persist(self, cosmos_client: ContainerProxy, blob_container_client: ContainerClient, blob_base_uri: str):
         if self.id is None or self.partition_key is None or self.resource is None:
             return
 
         resource_attributes = self.resource.get(SpanFieldName.ATTRIBUTES, None)
         if resource_attributes is None:
             return
 
+        if self.events and blob_container_client is not None and blob_base_uri is not None:
```

Comment:
Not None check is not necessary for now, right?

Reply:
Yes. The check is just for defend..